src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joined = '';
+
+    if (this.length === 0) {
+      return joined;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i] === undefined || this[i] === null
+        ? ''
+        : this[i].toString();
```

Comment:
there's no need for explicit conversion
```suggestion
        : this[i];
```